### PR TITLE
React routes task

### PR DESCRIPTION
**Pull request by Moraheem98**
- implemented react-router into project
- Created 2 pages (moviepage, homepage)
- created useEffect hook which is using thunk functions to dispatch actions based on the ID of a movie and display into MovieDetailsBanner

### CODE REVIEW

File: task-2/src/components/ErrorDisplay404/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+// import { Link } from 'react-router-dom';
+
+export const ErrorDisplay404 = () => {
+	return (
+		<div>
+			<p>Page Not Found</p>
+			<p>404</p>
+			{/* <Link to='/'>
```
Comment (Dmitry-White) [inline task-2/src/components/ErrorDisplay404/index.js]: It's a good functionality, let's uncomment it.
A side note - avoid leaving commented code without a GOOD reason in production.

File: task-2/src/components/Navigation/index.js [hunk 2]
```diff
@@ -11,7 +12,9 @@ export const Navigation = ({ setOpenModal }) => {
 		<div className='navigationContainer'>
 			<p style={{ color: '#F65261' }}>netflixroulette</p>
 			<div className='btnContainer'>
-				<SearchButton />
+				<Link to='/'>
+					<SearchButton />
```
Comment (Dmitry-White) [inline task-2/src/components/Navigation/index.js]: Couple of points:
- take a look inside SearchButton - you already have a "<Link>" there, wrapping SearchButton in Link again is redundant an prone to errors
- refactor you button component completely - make it just 1 component that accepts "text" as a prop and an optional "url" as prop (if "url" is truthy, wrap button contents in a Link, if falsy - render just button contents)
- go through the up and reuse your Button component where possible

File: task-2/src/components/SearchBar/uselocationTest.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// import React from 'react';
```
Comment (Dmitry-White) [inline task-2/src/components/SearchBar/uselocationTest.js]: Consider NOT leaving redundant commented code in your app.

**Review by Dmitry-White — CHANGES_REQUESTED** [review]
Consider doing some additional refactoring of the project.
Main focus - redundant code and buttons improvement.
Apart from the mentioned buttons in the comments below, your Formik component does have unstyled buttons that should look like the rest of the buttons in the project - do make use of the Button component there as well.

File: task-2/src/components/ErrorDisplay404/index.js (hunk 1 above)
Comment (Moraheem98) [reply task-2/src/components/ErrorDisplay404/index.js]: Resolved.

File: task-2/src/components/SearchBar/uselocationTest.js (hunk 3 above)
Comment (Moraheem98) [reply task-2/src/components/SearchBar/uselocationTest.js]: Resolved.

File: task-2/src/components/Navigation/index.js (hunk 2 above)
Comment (Moraheem98) [reply task-2/src/components/Navigation/index.js]: resolved.

**Review by Dmitry-White — CHANGES_REQUESTED** [review]
Considering the task is testing, we'd better resolve som of the issues in the app that are pending for quite some time now. Otherwise tests would be hard and would require updates (since you've started them already, you'll soon find out what I mean :) ).
DO DOUBLE CHECK how the provided design looks like and change your UI accordingly.
The following are the major changes that we need to introduce:

Main Page Top:
Screenshot 2022-04-04 at 20 55 55
1. Update your title and favicon
2. Get rid of the white gap between your app and browser window
3. Add some visual spacing between buttons. 
DO REMEMBER that changes applied to Button component would be reflected everywhere Button is used, so it might be better to address spacing outside Button.
4. Update input text colour for when user is typing something, at the moment it makes text almost invisible.
5. Add some spacing between Input and Button
6. Add some visual distinction between Tabs and the background - it's not clear what are those "All Action ... Romance" strings are supposed to do in the middle of the screen for UI/UX percpective
7. Update the dropdown visuals - there're many Codepen vanilla css solutions you can google and reuse, no need to make things look like 90s :)

Main Page Movie List:
Screenshot 2022-04-04 at 20 59 26
1. Get rid of the white gap between your app and browser window
2. Add an image url check to you card component and display a placeholder image if the movie image is missing
3. Either consult with the provided design or google a typical movie list design to improve UI of your cards (e.g title, buttons). You might use icon buttons instead of buttons with text to convey the meaning "REMOVE" or "EDIT"
4. Typo
5. Update buttons spacing and placement on the card. Additionally, double check whether buttons do work - it seems that "Remove" functionality has issues because of the "movie" destructuring.

Main Page Add Movie modal:
Screenshot 2022-04-04 at 21 02 59
1. Update modal title, lowercase and the placement don't look good at the moment
2. Close button is distorted and should be updated (to be an icon for example). Moreover, the placement needs to be updated as well.
3. The size and the placement of the modal is not appropriate. Not only it doesn't correspond to the content, but also it overflows beyond the screen.
4. Update the placement of the content
5. Update buttons spacing and placement on the modal
6. Double check the ability to Submit a new movie. It seems at the moment that API throws "400 Bad Request" because of the "genre" content.

Movie Page Edit Movie modal:
Screenshot 2022-04-04 at 21 05 52
1. Update the text logic in the header button - it should change according to the route change
2. Update buttons, their spacing and placement on the modal
3. Double check the ability to Submit an edited movie. It seems at the moment that your frontend validation fails because of the "genre" content.

**Review by Dmitry-White — APPROVED** [review]
Alright, this was a long one!
But now the code and the UI itself looks much cleaner and coherent, nice work!

**Pull request merged**